File: sky/org.kevoree.library.sky.api/src/main/scala/org/kevoree/library/sky/api/command/StopNodeCommand.scala
package org.kevoree.library.sky.api.command

import org.kevoree.library.sky.api.KevoreeNodeManager
import org.kevoree.ContainerRoot
import org.kevoree.api.PrimitiveCommand

/**
 * User: Erwan Daubert - erwan.daubert@gmail.com
 * Date: 25/06/13
 * Time: 12:02
 *
 * @author Erwan Daubert
 * @version 1.0
 */
class StopNodeCommand(iaasModel: ContainerRoot,targetChildName : String,nodeManager : KevoreeNodeManager) extends PrimitiveCommand {
  def execute () : Boolean = {
    //TODO PRUNE MODEL

    nodeManager.stopNode(iaasModel,targetChildName)
  }

  def undo () {
    nodeManager.startNode(iaasModel,targetChildName, iaasModel)
  }

  override def toString:String ={
    "AddNode " + targetChildName
  }
}

File: sky/org.kevoree.library.sky.api/src/main/java/org/kevoree/library/sky/api/execution/command/StopNodeCommand.java
package org.kevoree.library.sky.api.execution.command;

import org.kevoree.ContainerRoot;
import org.kevoree.api.PrimitiveCommand;
import org.kevoree.library.sky.api.execution.KevoreeNodeManager;

import java.util.Map;

/**
 * User: Erwan Daubert - erwan.daubert@gmail.com
 * Date: 25/06/13
 * Time: 12:02
 *
 * @author Erwan Daubert
 * @version 1.0
 */
public class StopNodeCommand implements PrimitiveCommand {

    private ContainerRoot iaasModel;
    private String targetChildName;
    private KevoreeNodeManager nodeManager;

    private Map<String, Object> registry;

    public StopNodeCommand(ContainerRoot iaasModel, String targetChildName, KevoreeNodeManager nodeManager, Map<String, Object> registry) {
        this.iaasModel = iaasModel;
        this.targetChildName = targetChildName;
        this.nodeManager = nodeManager;
        this.registry = registry;
    }

    public boolean execute() {
        //TODO PRUNE MODEL

        return nodeManager.stopNode(iaasModel, targetChildName);
    }

    public void undo() {
        nodeManager.startNode(iaasModel, targetChildName, iaasModel);
    }

    public String toString() {
        return "stopNode " + targetChildName;
    }
}
